pinot-controller/src/main/java/org/apache/pinot/controller/recommender/realtime/provisioning/MemoryEstimator.java

```diff
@@ -430,44 +422,48 @@ private long calculateMemoryForCompletedSegmentsPerPartition(long completedSegme
     return _numSegmentsQueriedPerHost;
   }
 
+  public void cleanup() {
+    FileUtils.deleteQuietly(_workingDir);
+  }
+
```

Comment:
MemoryEstimation should internally be responsible for doing the cleanup before returning. The caller need not call this